src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,10 @@
+package duke;
+
+/**
+ * Exceptions that may happen while using <code>Duke</code>.
+ */
+public class DukeException extends Exception {
```

Comment:
This is a good attempt at customizing exceptions.
In your TP, you may want to think about making them specific rather than just a single exception. (E.g., one for file IO, one for data formatting etc.,)